src/main.rs

```diff
@@ -168,6 +170,30 @@ where command is one of:
 
                 return
             }
+            "get_block_height" => {
+                if argv.len() < 4 {
+                    eprintln!("Usage: {} local get_block_height [vm-state.db]", argv[0]);
+                    process::exit(1);
+                }
+
+                let mut db = match ContractDatabaseConnection::open(&argv[3]) {
+                    Ok(db) => db,
+                    Err(error) => {
+                        eprintln!("Could not open vm-state: \n{}", error);
+                        process::exit(1);
+                    }
+                };
+
+                let mut sp = db.begin_save_point();
```

Comment:
This should be unnecessary.  You should be able to safely query the DB without setting up a save point.

Reply:
`begin_save_point()` is currently the only entry point to getting a `ContractDatabase` instance which is what implements the traits that perform various queries, but requires a `savepoint` instance. 
```rs
pub struct ContractDatabase <'a> {
    savepoint: Savepoint<'a>
}
```

There is some additional odd sqlite connection interplay between the ContractDatabase and AnalysisDatabase structs. Some of the comments and recent git history around these suggest its under dev right now, so I didn't want to perform any refactoring just yet. @kantai any points here?

Reply:
Yes, as of now, the only way to query the database is with a save point reference.

If we want a read-only database context, we should have a new struct for that, since I think it’s better to have Rust ensure that an interpreter execution only ever writes through save points. I don't see a strong motivation for this, though, since these "read only" database connections would never be in the critical path for any transaction validation.